### ISSUE TITLE
Lecture position link

### ISSUE THREAD

**Issue by acoollevel** [acoollevel/uoa-mediaplayer-plus#37]
Allow people to link to a specific part of a lecture ie. `my-lecture-url?t=1m50s`

**Comment by LRTNZ** [comment]
Hmmm, do we do t=1m50s, or just seconds?

**Comment by acoollevel** [comment]
Unfortunately when you include a hash or GET parameters in the URL, the media player fails to load properly. This is not something we can fix, as people without the extension need to be able to use these links as well. What we'll likely have to do is proxy these links through our own server instead.

**Comment by LRTNZ** [comment]
Ok, so how do we get them to land on the mediaplayer page, while still being able to pass some information?
Maybe do something like:
1. Direct to proxy address on our server
2. Proxy page talks to our plugin, primes it with a hash, and gives it the link to direct to.
3. Page gets redirects to the actual page for the video.
4. Plugin sees it has been primed with a hash for the video, and makes a call to our server to get a data structure with information relating to that hash?

**Comment by acoollevel** [comment]
> Proxy page talks to our plugin, primes it with a hash, and gives it the link to direct to.
Because the player crashes when using a hash, I would avoid using them

**Comment by LRTNZ** [comment]
> 
> 
> > Proxy page talks to our plugin, primes it with a hash, and gives it the link to direct to.
> 
> Because the player crashes when using a hash, I would avoid using them

No, as in it does not get injected into the URL. It gets put into our plugins local storage. Like a specific part that is only for "When the plugin next is loaded on a mediaplayer link, try talking to the server with this hash to see what details there are relating to the playback". So basically a detour via our proxy link that passes some information to our plugin, that is then used when the video playback page is then loaded. This means we do not have to interfere with the URL for the video itself.

**Comment by LRTNZ** [comment]
So basically:
1. User gets directed to a link, that goes to our server. That page talks to the plugin, and passes it a hash/key it can use to call back to the server to get some information. This can also have logic included to direct users to download our plugin and such, if we cannot talk to it.
2. The plugin/page then redirects to the standard vanilla URL for the video in question.
3. Once that page has loaded, the plugin can take the hash that it has stored for use the next time it loads a video, and can make a get request to the server with that value. 
4. The server can then pass back the required information to the plugin, that can manipulate the player as required.

**Comment by acoollevel** [comment]
> call back to the server to get some information

What information? The lecture position time can just be included in the URL, right?

**Comment by LRTNZ** [comment]
> 
> Unfortunately when you include a hash or GET parameters in the URL, the media player fails to load properly. This is not something we can fix, as people without the extension need to be able to use these links as well. What we'll likely have to do is proxy these links through our own server instead.
>
I thought this was an issue, that meant we could not do the stuff in the media URL?
> 
> What information? The lecture position time can just be included in the URL, right?

**Comment by acoollevel** [comment]
Put it in the proxy URL

**Comment by LRTNZ** [comment]
> 
> 
> Put it in the proxy URL

I was thinking something like that - but we still need to get it to the player when we direct them to the mediaplayer link right? I guess the plugin will have to extract that info from the link, and then store that when the page is redirected?
